databuilder/dsl.py

```diff
@@ -258,34 +258,16 @@ def __init__(self, value: Value | Comparator):
     def _is_comparator(self) -> bool:
         return isinstance(self.value, Comparator)
 
-    def __gt__(self, other: ValueExpression) -> PatientSeries:
-        return PatientSeries(value=self.value > other)
-
-    def __ge__(self, other: ValueExpression) -> PatientSeries:
-        return PatientSeries(value=self.value >= other)
-
-    def __lt__(self, other: ValueExpression) -> PatientSeries:
-        return PatientSeries(value=self.value < other)
-
-    def __le__(self, other: ValueExpression) -> PatientSeries:
-        return PatientSeries(value=self.value <= other)
-
-    def __eq__(self, other: ValueExpression) -> PatientSeries:  # type: ignore[override]
-        # All python objects have __eq__ and __ne__ defined, so overriding these method s
+    def __eq__(self, other: ValueExpression) -> BoolSeries:  # type: ignore[override]
+        # All python objects have __eq__ and __ne__ defined, so overriding these methods
         # involves overriding them on a superclass (`object`), which results in
         # a typing error as it violates the violates the Liskov substitution principle
         # https://mypy.readthedocs.io/en/stable/common_issues.html#incompatible-overrides
         # We are deliberately overloading the operators here, hence the ignore
-        return PatientSeries(value=self.value == other)
+        return BoolSeries(value=self.value == other)
 
-    def __ne__(self, other: ValueExpression) -> PatientSeries:  # type: ignore[override]
-        return PatientSeries(value=self.value != other)
-
-    def __and__(self, other: PatientSeries) -> PatientSeries:
-        return PatientSeries(value=self.value & other.value)
-
-    def __or__(self, other: PatientSeries) -> PatientSeries:
-        return PatientSeries(value=self.value | other.value)
+    def __hash__(self) -> int:
+        return hash(repr(self.value))
```

Comment:
I know these are alphabetised, but I wonder whether it'd be easier to understand the class if the DSL methods (`__eq__` and `__invert__`) were grouped separately from the other dunder methods (eg `__hash__` and `__repr__`).

Reply:
I tried a variety of ways to order these and eventually just gave up and fell back to alphetising them.  For instance: `__hash__` and `__eq__` are intertwined because of the [way `__hash__` works](https://docs.python.org/3.10/reference/datamodel.html#object.__hash__).

Reply:
Fine, not important.